server/mscalendar/availability.go

```diff
@@ -26,42 +26,50 @@ const (
 	logTruncateLimit                = 5
 )
 
+type StatusSyncJobSummary struct {
+	NumberOfUserErrorInStatusChange int
+	NumberOfUserStatusChange        int
+	NumberOfUserProcessed           int
+}
+
 type Availability interface {
 	GetCalendarViews(users []*store.User) ([]*remote.ViewCalendarResponse, error)
-	Sync(mattermostUserID string) (string, error)
-	SyncAll() (string, error)
+	Sync(mattermostUserID string) (string, *StatusSyncJobSummary, error)
+	SyncAll() (string, *StatusSyncJobSummary, error)
 }
 
-func (m *mscalendar) Sync(mattermostUserID string) (string, error) {
+func (m *mscalendar) Sync(mattermostUserID string) (string, *StatusSyncJobSummary, error) {
 	user, err := m.Store.LoadUserFromIndex(mattermostUserID)
 	if err != nil {
-		return "", err
+		return "", nil, err
 	}
 
 	return m.syncUsers(store.UserIndex{user})
 }
 
-func (m *mscalendar) SyncAll() (string, error) {
+func (m *mscalendar) SyncAll() (string, *StatusSyncJobSummary, error) {
 	err := m.Filter(withSuperuserClient)
 	if err != nil {
-		return "", errors.Wrap(err, "not able to filter the super user client.")
+		return "", nil, errors.Wrap(err, "not able to filter the super user client.")
 	}
 
 	userIndex, err := m.Store.LoadUserIndex()
 	if err != nil {
 		if err.Error() == "not found" {
-			return "No users found in user index", nil
+			return "No users found in user index", nil, nil
```

Comment:
maybe this `No user was found in the user index` or this `No users were found in the user index`?

Reply:
all these messages exists before hand so i think we should not change it as they are beyond its scope